nixos/modules/programs/x2goserver.nix

```diff
@@ -0,0 +1,121 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.x2goserver;
+
+  defaults = {
+    log = { "loglevel" = cfg.logLevel; };
+    x2goagent = { "port_randomization" = "pure-random"; };
+    superenicer = { "enable" = "no"; };
+    telekinesis = { "enable" = "no"; };
+  };
+  confText = generators.toINI {} (recursiveUpdate defaults cfg.config);
+  x2goServerConf = pkgs.writeText "x2goserver.conf" confText;
+
+  x2goAgentOptions = pkgs.writeText "x2goagent.options" ''
+    X2GO_NXOPTIONS=""
+    X2GO_NXAGENT_DEFAULT_OPTIONS="${concatStringsSep " " cfg.nxagentDefaultOptions}"
+  '';
+
+in {
+  options.programs.x2goserver = {
+    enable = mkEnableOption "x2goserver";
+
+    logLevel = mkOption {
+      type = types.str;
+      default = "notice";
+      description = ''
+        The log level x2goserver will log at.
+        Possible levels are: emerg, alert, crit, err, warning, notice, info, debug
+      '';
+    };
+
+    nxagentDefaultOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ "-extension GLX" "-nolisten tcp" ];
+      example = [ "-extension GLX" "-nolisten tcp" ];
+      description = ''
+        List of default nx agent options.
+      '';
+    };
+
+    config = mkOption {
+      type = types.attrsOf types.attrs;
+      default = {};
+      description = "x2goserver.conf ini configuration as nix attributes.";
+      example = literalExample ''
+        superenicer = {
+          "enable" = "yes";
+          "idle-nice-level" = 19;
+        };
+        telekinesis = { "enable" = "no"; };
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups.x2go = {};
+    users.users.x2go = {
+      createHome = true;
+      home = "/var/lib/x2go";
+      group = "x2go";
+    };
+
+    security.wrappers.x2gosqliteWrapper = {
+      source = "${pkgs.x2goserver}/lib/x2go/libx2go-server-db-sqlite3-wrapper.pl";
+      owner = "root";
```

Comment:
Why `root`? If this is really needed a comment should explain it here. Same for the print wrapper.

Reply:
This is only setgid so I don't think there is a risk, and that's what the default distribution does everywhere. But to be safe I've changed it to user x2go as well, so let's hope nothing breaks.